demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -119,3 +120,154 @@ InferenceEngine::BlobMap Cnn::Infer(const cv::Mat &frame) {
+
+    channels_ = input_info->getTensorDesc().getDims()[1];
+    input_size_ = cv::Size(input_info->getTensorDesc().getDims()[3], input_info->getTensorDesc().getDims()[2]);
+
+    // --------------------------- Loading model to the device -------------------------------------------
+    ExecutableNetwork executable_network_encoder = ie.LoadNetwork(network_encoder, deviceName);
+    ExecutableNetwork executable_network_decoder = ie.LoadNetwork(network_decoder, deviceName);
+    // ---------------------------------------------------------------------------------------------------
+
+    // --------------------------- Creating infer request ------------------------------------------------
+    infer_request_encoder_ = executable_network_encoder.CreateInferRequest();
+    infer_request_decoder_ = executable_network_decoder.CreateInferRequest();
+    // ---------------------------------------------------------------------------------------------------
+
+    is_initialized_ = true;
+}
+
+void EncoderDecoderCNN::setInOutNames(const std::string out_enc_hidden_name,
+                                    const std::string out_dec_hidden_name,
+                                    const std::string in_dec_hidden_name,
+                                    const std::string features_name,
+                                    const std::string in_dec_symbol_name,
+                                    const std::string out_dec_symbol_name,
+                                    const std::string logits_name) {
+    out_enc_hidden_name_ = out_enc_hidden_name;
+    out_dec_hidden_name_ = out_dec_hidden_name;
+    in_dec_hidden_name_ = in_dec_hidden_name;
+    features_name_ = features_name;
+    in_dec_symbol_name_ = in_dec_symbol_name;
+    out_dec_symbol_name_ = out_dec_symbol_name;
+    logits_name_ = logits_name;
+}
+
+InferenceEngine::BlobMap EncoderDecoderCNN::Infer(const cv::Mat &frame) {
+    std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+    /* Resize manually and copy data from the image to the input blob */
+    InferenceEngine::LockedMemory<void> inputMapped =
+        InferenceEngine::as<InferenceEngine::MemoryBlob>(infer_request_encoder_.GetBlob(input_name_))->wmap();
+    float* input_data = inputMapped.as<float *>();
+
+    cv::Mat image;
+    if (channels_ == 1) {
+         cv::cvtColor(frame, image, cv::COLOR_BGR2GRAY);
+    } else {
+        image = frame.clone();
+    }
+
+    image.convertTo(image, CV_32F);
+    cv::resize(image, image, input_size_);
+
+    int image_size = input_size_.area();
+
+    if (channels_ == 3) {
+        for (int pid = 0; pid < image_size; ++pid) {
+            for (int ch = 0; ch < channels_; ++ch) {
+                input_data[ch * image_size + pid] = image.at<cv::Vec3f>(pid)[ch];
+            }
+        }
+    } else if (channels_ == 1) {
+        for (int pid = 0; pid < image_size; ++pid) {
+            input_data[pid] = image.at<float>(pid);
+        }
+    }
```

Comment:
Thanks, done. But I still have to store `input_name_`, otherwise I can not access the blob. Maybe I'm doing something wrong.

Reply:
`matU8ToBlob()` doesn't require `blob` arg to be non const. Mark it as `const`. This will allow to pass temporary object.
`image = frame.clone();` - no need to copy image.

Reply:
Done, but `input_name_` still is needed.

Reply:
Done